File: #04/q14 - deletee/deletee.js
deletee = (x, arr) => {
    arr.splice(arr.findIndex(f = val => val == x),1);
    return arr;
}

console.log(deletee(5, []))
console.log(deletee(1, [1]))
console.log(deletee(4, [1,3,4]))
console.log(deletee(3, [4,3,1,3]))
console.log(deletee(5, [1,5,6,9]))

File: #04/q14 - deletee/deletee.hs

deletee :: Int -> [Int] -> [Int]
deletee a [] = []
deletee a (x:xs)
    | a == x = xs
    | otherwise = x : deletee a xs

main :: IO ()
main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ deletee a b